src/workerd/api/node/tests/dns-nodejs-test.js

```diff
@@ -37,3 +36,41 @@ export const errorCodesExist = {
     strictEqual(typeof dnsPromises.NODATA, 'string');
   },
 };
+
+export const resolveTxt = {
+  async test() {
+    // An idiomatic test that avoids checking for a specific value
+    function testResults(results) {
+      ok(Array.isArray(results));
+      ok(results.length > 0);
+      ok(
+        results.find((r) => r.find((v) => v.includes('spf1'))),
+        'One of the entries should have spf1'
+      );
+      ok(
+        results.at(0).at(0)[0] !== '"',
+        'First character of a TXT record should not be a quotation mark'
+      );
+    }
+
+    {
+      // Promise API
+      const results = await dnsPromises.resolveTxt('cloudflare.com');
+      testResults(results);
+    }
+
+    {
+      // Callback API
+      const { promise, resolve, reject } = Promise.withResolvers();
+      dns.resolveTxt('cloudflare.com', (error, results) => {
+        if (error) {
+          reject(error);
+          return;
+        }
+        testResults(results);
+        resolve();
+      });
+      await promise;
```

Comment:
```suggestion
      // Callback API
      dns.resolveTxt('cloudflare.com', (error, results) => {
        if (error) {
          return fail(error);
        }
        testResults(results);
      });
```

Reply:
@jasnell Does this hold, or do we need Promise.withResolvers() to wait for the callback response?

Reply:
The use of `Promise.withResolvers()` looks correct here.